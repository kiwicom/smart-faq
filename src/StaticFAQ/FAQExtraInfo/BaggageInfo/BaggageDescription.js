// @flow

import * as React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { BaggageSmall, BaggageBig } from '@kiwicom/orbit-components/lib/icons';

import type { BaggageDescription as BaggageDescriptionProps } from './__generated__/BaggageDescription.graphql';

type Props = {|
  type: 'Cabin' | 'Checked',
  data: BaggageDescriptionProps,
|};

function renderIcon(type: string) {
  switch (type) {
    case 'Cabin':
      return <BaggageSmall size="medium" customColor="#bac7d5" />;
    case 'Checked':
      return <BaggageBig size="medium" customColor="#bac7d5" />;
  }
  return null;
}

const BaggageDescription = ({
  type,
  data: { height, weight, width, length },
}: Props) => {
  return (
    <React.Fragment>
      <div className="baggageRow">
        <div className="baggageNumber">
          <p>{1}x</p>
        </div>
        {renderIcon(type)}
        <p className="baggageWeight">
          {type} baggage {weight} kg
        </p>
        <div className="baggageSize">
          <p>
            {height} x {width} x {length}
          </p>
        </div>
      </div>
      <hr className="separationLine" />
      <style jsx>
        {`
          div.baggageRow {
            padding: 15px 24px 15px 24px;
          }
          div.baggageSize {
            display: inline-block;
            float: right;
          }
          div.baggageSize p {
            font-size: 12px;
            line-height: 1.4;
            color: #46515e;
          }
          div.baggageNumber {
            display: inline-block;
            margin-right: 8px;
          }
          div.baggageNumber p {
            font-size: 14px;
            font-weight: bold;
            line-height: 1.4;
            color: #46515e;
            display: inline-block;
          }
          p.baggageWeight {
            font-size: 14px;
            font-weight: bold;
            line-height: 1.4;
            color: #46515e;
            display: inline-block;
          }
          hr.separationLine {
            border: solid 1px #e8edf1;
            width: 100%;
          }
        `}
      </style>
    </React.Fragment>
  );
};

export default createFragmentContainer(
  BaggageDescription,
  graphql`
    fragment BaggageDescription on Baggage {
      height
      weight
      width
      length
    }
  `,
);
