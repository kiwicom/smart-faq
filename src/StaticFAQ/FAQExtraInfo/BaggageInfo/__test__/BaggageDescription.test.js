// @flow

import * as React from 'react';
import { shallow, render } from 'enzyme';

import { BaggageDescription } from '../BaggageDescription';

const mockRefType: any = null;
const baggage = {
  $refType: mockRefType,
  height: 40,
  width: 30,
  length: 15,
  weight: 10,
};

const baggageDescriptionComponent = (
  <BaggageDescription data={baggage} type="Checked" />
);

describe('BaggageDescription', () => {
  it('should render correctly', () => {
    expect(shallow(baggageDescriptionComponent)).toMatchSnapshot();
  });

  it('should render correct icon according to baggage type', () => {
    const component = shallow(baggageDescriptionComponent);
    expect(component.find('BaggageChecked').exists()).toBeTruthy();
  });

  it('should show size of the baggage', () => {
    const wrapper = render(baggageDescriptionComponent);
    expect(wrapper.text()).toContain('40 x 30 x 15 cm');
  });

  it('should show weight of the baggage', () => {
    const wrapper = render(baggageDescriptionComponent);
    expect(wrapper.text()).toContain('10 kg');
  });
});
