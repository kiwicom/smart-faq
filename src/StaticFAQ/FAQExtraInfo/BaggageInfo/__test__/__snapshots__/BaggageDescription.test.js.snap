// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaggageDescription should render correctly 1`] = `
<React.Fragment>
  <hr
    className="jsx-677876937 separationLine"
  />
  <div
    className="jsx-677876937 baggageRow"
  >
    <BaggageChecked
      customColor="#bac7d5"
      size="medium"
    />
    <p
      className="jsx-677876937 baggageWeight"
    >
      Checked
       baggage 
      10
       kg
    </p>
    <div
      className="jsx-677876937 baggageSize"
    >
      <p
        className="jsx-677876937"
      >
        40
         x 
        30
         x 
        15
         cm
      </p>
    </div>
  </div>
  <JSXStyle
    css="div.baggageRow.jsx-677876937{padding:15px 24px 15px 24px;}div.baggageSize.jsx-677876937{display:inline-block;float:right;}div.baggageSize.jsx-677876937 p.jsx-677876937{font-size:12px;line-height:1.4;color:#46515e;}div.baggageNumber.jsx-677876937{display:inline-block;margin-right:8px;}div.baggageNumber.jsx-677876937 p.jsx-677876937{font-size:14px;font-weight:bold;line-height:1.4;color:#46515e;display:inline-block;}p.baggageWeight.jsx-677876937{margin-left:5px;font-size:14px;font-weight:bold;line-height:1.4;color:#46515e;display:inline-block;}hr.separationLine.jsx-677876937{height:1px;background-color:#e8edf1;border:none;}"
    styleId="677876937"
  />
</React.Fragment>
`;
